main.tf

```diff
@@ -1,136 +1,179 @@
+#   aws_key_name            = "${module.dcos.infrastructure.aws_key_name}"
+#   security_group_admin    = "${module.dcos.infrastructure.security_groups.admin}"
+#   security_group_internal = "${module.dcos.infrastructure.security_groups.internal}"
+#   bootstrap_private_ip    = "${module.dcos.infrastructure.bootstrap.private_ip}"
+#   bootstrap_public_ip     = "${module.dcos.infrastructure.bootstrap.public_ip}"
+#   bootstrap_os_user       = "${module.dcos.infrastructure.bootstrap.os_user}"
+#   ssh_private_key_file    = "~/.ssh/id_rsa"
+#   masters_private_ips     = "${module.dcos.infrastructure.masters.private_ips}"
+# }
+
+# ```
+
+
+data "aws_ami" "winAmi" {
+ owners                               = ["amazon"]
+  filter{
+    name                              = "name"
+    values                            = ["Windows_Server-1809-English-Core-ContainersLatest-2019.05.15"]
+  }
 }
 
-module "dcos-tested-oses" {
-  source  = "dcos-terraform/tested-oses/aws"
-  version = "~> 0.2.0"
-
-  providers = {
-    aws = "aws"
+resource "aws_security_group" "rdp" {
+  description                         = "Allow incoming rdp traffic from admin_ips"
+  vpc_id                              = "${var.vpc_id}"
+
+      tags = {
+        Name                          = "allow_rdp"
+      }
+      ingress {
+        from_port                     = 3389
+        to_port                       = 3389
+        protocol                      = "tcp"
+      cidr_blocks                     = ["${var.admin_ips}"]
+            }
+      ingress {
+        from_port                     = 3389
+        to_port                       = 3389
+        protocol                      = "udp"
+      cidr_blocks                     = ["${var.admin_ips}"]
+            }
+      ingress {
+        from_port                     = 0
+        to_port                       = 65535
+        protocol                      = "tcp"
+      cidr_blocks                     = ["0.0.0.0/0"]
+            }
+    }
+
+resource "aws_instance" "mesos-agent-windows" {
+  count                               = "${var.num_winagent}"
+  ami                                 = "${data.aws_ami.winAmi.id}"
+  instance_type                       = "t3.xlarge"
+  get_password_data                   = "true"
+  root_block_device {
+    volume_type                       = "gp2"
+    volume_size                       = 230
+    }
+  tags = {
+    Name                              = "${var.cluster_name}-winagent-${count.index}"
+    expiration                        = "${var.expiration}"
+    owner                             = "${var.owner}"
   }
-
-  os = "${var.dcos_instance_os}"
+  key_name                            = "${var.aws_key_name}"
+  subnet_id                           = "${element(var.subnet_id,0)}"
+  security_groups                     = ["${list(
+      var.security_group_internal,
+      var.security_group_admin,
+      aws_security_group.rdp.id)}"]
+
+  user_data                           = <<-EOF
```

Comment:
the user should still be able to overwrite this behavior. I also would suggest either using local variables or a file for this script which makes it more readable